src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -168,6 +166,8 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
             // Ask the metadata manager if we have any dependencies due to reflectability.
             factory.MetadataManager.GetDependenciesDueToReflectability(ref dependencyList, factory, _type);
 
+            CodeBasedDependencyAlgorithm.AddInterestingInteropConstructedTypeDependencies(ref dependencyList, factory, _type);
```

Comment:
Right. We don't only add dependencies for the delegates on this code path . So we should be fine for now.